plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/IcebergGlueCatalogModule.java

```diff
@@ -56,10 +56,11 @@ protected void setup(Binder binder)
         binder.bind(TrinoCatalogFactory.class).to(TrinoGlueCatalogFactory.class).in(Scopes.SINGLETON);
         newExporter(binder).export(TrinoCatalogFactory.class).withGeneratedName();
 
-        install(conditionalModule(
-                IcebergGlueCatalogConfig.class,
-                IcebergGlueCatalogConfig::isSkipArchive,
-                internalBinder -> newSetBinder(internalBinder, RequestHandler2.class, ForGlueHiveMetastore.class).addBinding().toInstance(new SkipArchiveRequestHandler())));
+        GlueHiveMetastoreConfig glueHiveConfig = buildConfigObject(GlueHiveMetastoreConfig.class);
+        IcebergGlueCatalogConfig glueIcebergConfig = buildConfigObject(IcebergGlueCatalogConfig.class);
+        if (glueIcebergConfig.isSkipArchive() || !glueHiveConfig.isSkipArchive()) {
```

Comment:
Not quite clear what `!glueHiveConfig.isSkipArchive()` means.
Why do we care about hive setting here?

Reply:
This is to avoid binding the same Guice dependency multiple times.

Reply:
I don't quite follow. Both settings relate to the same thing.

Why would we check for such a setting constellation?
```
iceberg.glue.skip-archive=true
hive.metastore.glue.skip-archive=false
```

Maybe we should rather add  `hive.metastore.glue.skip-archive` and use  `@LegacyConfig("iceberg.glue.skip-archive")`

Reply:
@nineinchnick Let's move to `@DefunctConfig` as we talked offline. `iceberg.glue.skip-archive` is enabled by default and I suppose users don't change this property.

Reply:
Done! PTAL